pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt

```diff
@@ -2,159 +2,197 @@ package gov.cdc.ocio.processingstatusapi.plugins
-     */
-    suspend fun validate(receivedMessages: ReceiveMessageResponse) {
-        try {
-            receivedMessages.messages?.forEach { message ->
-                SchemaValidation.logger.info("Received message from AWS SQS: ${message.body}")
-                val awsSQSProcessor = AWSSQSProcessor()
-                message.body?.let {
-                    awsSQSProcessor.processMessage(it)
+            /**
+             * Deletes messages from AWS SQS Service that has been validated
+             * @param receivedMessages the list of message(s) received from the queue to be deleted
+             * @throws Exception
+             */
+            suspend fun deleteMessage(receivedMessages: ReceiveMessageResponse) {
+                receivedMessages.messages?.forEach { message ->
+                    try {
+                        retryWithBackoff(numOfRetries = 5) {
+                            val deleteMessageRequest = DeleteMessageRequest {
+                                this.queueUrl = queueUrl
+                                this.receiptHandle = message.receiptHandle
+                            }
+                            sqsClient.deleteMessage(deleteMessageRequest)
+                        }
+                        SchemaValidation.logger.info(
+                                "Successfully deleted processed report from AWS SQS"
+                        )
+                    } catch (e: Exception) {
+                        SchemaValidation.logger.error(
+                                "Something went wrong while deleting the report from the queue ${e.message}"
+                        )
+                    }
                 }
             }
-            deleteMessage(receivedMessages)
-        }catch (e: Exception) {
-            SchemaValidation.logger.error("An Exception occurred during validation ${e.message}")
-            deleteMessage(receivedMessages)
-        }
-    }
-    /**
-     * The `consumeMessages` function continuously listens for and processes messages from an AWS SQS queue.
-     * This function runs in a non-blocking coroutine, retrieving messages from the queue, validating them using
-     * `AWSSQSProcessor`, and then deleting the processed messages from the queue.
-     *
-     * @throws Exception
-     * @throws AwsServiceException
-     */
-    fun consumeMessages() {
-        SchemaValidation.logger.info("Consuming messages from AWS SQS")
-        CoroutineScope(Dispatchers.IO).launch {
-            while (true) {
-                var receivedMessages: ReceiveMessageResponse?
+            /**
+             * Validates messages from the AWS SQS Service
+             * @param receivedMessages the list of message(s) received from the queue to be
+             * validated
+             * @throws Exception thrown during validation `and` it's important to delete the message
+             * as it will be persisted to dead-letter container
+             * ```
```

Comment:
Looks like you ran an auto-reformat on the code and it something funky with this comment formatting.  See the three ticks?

Reply:
I'm going to try and get rid of the formatting changes, sry editor did it without me noticing